src/main/java/com/github/jtama/crazy/dop/PrettyPrinter.java

```diff
@@ -15,14 +15,29 @@ public static String toString(List<PlayingCard> cards) {
 
 
     public static String toString(PlayingCard playingCard) {
-        return switch (playingCard) {
-            case TrumpCard(Integer idx) -> "The trump n°%s is strong".formatted(idx);
-            case NumberSuitCard suitCard when suitCard.index() == 1 -> "The first of %s(%s) is very weak".formatted(suitCard.color().name().toLowerCase(), suitCard.color().getSymbol());
-            case NumberSuitCard suitCard when suitCard.index() == 2 -> "The second of %s(%s) is very weak".formatted(suitCard.color().name().toLowerCase(), suitCard.color().getSymbol());
-            case NumberSuitCard suitCard when suitCard.index() == 3 -> "The third of %s(%s) is very weak".formatted(suitCard.color().name().toLowerCase(), suitCard.color().getSymbol());
-            case NumberSuitCard suitCard when suitCard.index() < 7 -> "The %sth of %s(%s) is still weak".formatted(suitCard.index(),suitCard.color().name().toLowerCase(), suitCard.color().getSymbol());
-            case NumberSuitCard(Color color, Integer index) -> "The %sth of %s(%s) may win you a hand".formatted(index, color.name().toLowerCase(), color.getSymbol());
-            case RoyalSuitCard(Color color, Face face) -> "The %s of %s(%s) is strong".formatted(face.displayName(), color.name().toLowerCase(), color.getSymbol());
-        };
+        if(playingCard instanceof SuitCard suitCard){
```

Comment:
I think that's already pattern matching in the initial state. I'd rather make en initial state that is 100% java 8 compatible.

What do you think ?

Reply:
If I have correctly grasped the flow of the workshop, we should start by creating a branch called 'start' that utilizes classic Java 8. The initial step will involve replacing all records with traditional Java classes. Next, we will introduce the concept of records and demonstrate how they can enhance code clarity. Finally, in the PrettyPrinter class, we will revert the conditional logic to a standard if statement and apply pattern matching to reach the desired outcome.

Reply:
That's exactly what's expected. And the start branch should be as ugly as possible.
And contain the documentation with eache expected steps